src/Surfnet/StepupRa/RaBundle/Security/Authentication/Token/SamlToken.php

```diff
@@ -61,6 +63,34 @@ public function getInstitutionConfigurationOptions()
     }
 
     /**
+     * @param string $institution
+     * @param InstitutionConfigurationOptions $institutionConfigurationOptions
+     */
+    public function changeInstitutionScope(
+        $institution,
+        InstitutionConfigurationOptions $institutionConfigurationOptions
+    ) {
+        if ($this->getUser() === null) {
+            throw new LogicException('Cannot change institution scope: token does not contain a user');
+        }
+
+        $roles = array_map(function (RoleInterface $role) {
+            return $role->getRole();
+        }, $this->getRoles());
+
+        if (!in_array('ROLE_SRAA', $roles)) {
+            throw new ForbiddenException(sprintf(
```

Comment:
Hmm, since the application doesn't handle this exception, and the exception is not part of the public API (as in a library), I see no use for a more specific "ForbiddenException" over the existing "RuntimeException". The exception message is already very clear. What does it add?